examples/imagefilter/node/main.ts

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env node
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: MIT
+
+import * as slint from "slint-ui";
+import Jimp from "jimp";
+
+class Filter {
+    name: string;
+    applyFunction: (image: slint.ImageData) => slint.ImageData;
+
+    constructor(
+        name: string,
+        applyFunction: (image: slint.ImageData) => slint.ImageData
+    ) {
+        this.name = name;
+        this.applyFunction = applyFunction;
+    }
+}
+
+class Filters extends slint.Model<string> {
+    #filters: Filter[];
+
+    constructor(filters: Filter[]) {
+        super();
+        this.#filters = filters;
+    }
+
+    at(index: number): Filter {
+        return this.#filters[index];
+    }
+
+    rowCount(): number {
+        return this.#filters.length;
+    }
+
+    rowData(row: number): string {
+        return this.#filters[row].name;
+    }
+
+    setRowData(row: number, data: string): void {
+        // not needed for this example
+        throw new Error("Method not implemented.");
+    }
+}
+
+const demo = slint.loadFile("../ui/main.slint") as any;
+const mainWindow = new demo.MainWindow();
+
+Jimp.read("../assets/cat.jpg").then((sourceImage) => {
+    mainWindow.original_image = sourceImage.bitmap;
+});
```

Comment:
I think promise chains should only be used at the end of functions, or unless it's explicit / obvious that there's no need to wait for the promise to resolve before continuing. In this case strictly speaking the promise must resolve before allowing the UI to become active, as otherwise the filter will be called on an empty image.

```suggestion
const sourceImage = await Jimp.read("../assets/cat.jpg");
mainWindow.original_image = sourceImage.bitmap;
```